### PR TITLE
[OTX][Hot-fix] Inst-seg, Det + Simple-Autoconfig test Hotfix

### PR DESCRIPTION
**Pull request by sungmanc**
# Summary
* Fix `detection_adapter.py` to resolve inst-seg, det issue: wrong implementation
* Fix `test_cli.py`: not use workspace in otx_root, use /tmp directory to save the workspace

### CODE REVIEW

File: otx/cli/utils/tests.py [hunk 1]
```diff
@@ -732,10 +732,13 @@ def otx_build_backbone_testing(root, backbone_args):
     ), f"{model_config['model']['backbone']['type']} != {backbone}"
 
 
-def otx_build_auto_config(otx_dir: str, args: Dict[str, str]):
+def otx_build_auto_config(root, otx_dir: str, args: Dict[str, str]):
+    workspace_root = os.path.join(root, "otx-workspace-CLASSIFICATION")
```
Comment (eunwoosh) [inline otx/cli/utils/tests.py]: Is this function executed only for classification?

Comment (sungmanc) [reply otx/cli/utils/tests.py]: For now, yes. I'll add for detection, segmentation later :)

Comment (eunwoosh) [reply otx/cli/utils/tests.py]: Then how about using general name to consider that?

Comment (sungmanc) [reply otx/cli/utils/tests.py]: It sounds good, thanks for opinion. --> Edited

**Review by eunwoosh — DISMISSED** [review]

File: otx/core/data/adapter/detection_dataset_adapter.py [hunk 2]
```diff
@@ -36,16 +36,18 @@ def get_otx_dataset(self) -> DatasetEntity:
                     image = Image(file_path=datumaro_item.media.path)
                     shapes = []
                     for ann in datumaro_item.annotations:
-                        if self.task_type is TaskType.DETECTION and ann.type == AnnotationType.polygon:
+                        if (
+                            str(self.task_type) is TaskType.INSTANCE_SEGMENTATION.name
```
Comment (cih9088) [inline otx/core/data/adapter/detection_dataset_adapter.py]: Why do we need `str(*)` and `*.name`?

Comment (harimkang) [reply otx/core/data/adapter/detection_dataset_adapter.py]: Can't we just use `self.task_type is TaskType.INSTANCE_SEGMENTATION`?

**Review by harimkang — DISMISSED** [review]
When I run it locally, `test_cli.py` seems to work fine in `/tmp`. Thanks for the correction.
Everything looks good except for the question about the modification with an if statement comparing str to name.

File: otx/core/data/adapter/detection_dataset_adapter.py (hunk 2 above)
Comment (harimkang) [reply otx/core/data/adapter/detection_dataset_adapter.py]: This part was corrected after discussing with sungman.
https://github.com/openvinotoolkit/training_extensions/pull/1572/commits/19c016e8cd4d03a38bb489e2d933c0724f108ebe

**Review by cih9088 — DISMISSED** [review]

**Review by cih9088 — APPROVED** [review]

**Review by harimkang — APPROVED** [review]

**Pull request merged**